pyquil/paulis.py

```diff
@@ -33,6 +33,9 @@
 from collections.abc import Sequence
 import warnings
 
+NumberT = Union[int, float, complex]
+PauliT = Union['PauliTerm', 'PauliSum', NumberT]
```

Comment:
I'd rather see this be `PauliDesignator = Union['PauliTerm', 'PauliSum']` to match the general framework that @appleby put together in `quilatom`, `quilbase`, and `gates`. Then it's just a simple `Union[PauliDesignator, Number]` for `PauliT` cases (and there are also some cases below where `PauliDesignator` alone is the type).